src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnectionOptions.cs

```diff
@@ -20,48 +21,110 @@ internal QuicConnectionOptions()
+    public long DefaultStreamErrorCode { get; set; } = -1;
+
+    /// <summary>
+    /// Error code used for <see cref="QuicConnection.CloseAsync(long, Threading.CancellationToken)"/> when the connection gets disposed.
+    /// To use different close error code, call  <see cref="QuicConnection.CloseAsync(long, Threading.CancellationToken)"/> explicitly before disposing.
+    /// </summary>
+    // QUIC doesn't allow negative value: https://www.rfc-editor.org/rfc/rfc9000.html#integer-encoding
+    // We can safely use this to distinguish if user provided value during validation.
+    public long DefaultCloseErrorCode { get; set; } = -1;
+
+    /// <summary>
+    /// Validates the options and potentially sets platform specific defaults.
+    /// </summary>
+    /// <param name="argumentName">Name of the from the caller.</param>
+    internal virtual void Validate(string argumentName)
+    {
+        if (MaxInboundUnidirectionalStreams < 0 || MaxInboundBidirectionalStreams > ushort.MaxValue)
+        {
+            throw new ArgumentOutOfRangeException($"{nameof(QuicConnectionOptions.MaxInboundBidirectionalStreams)} should be within [0, {ushort.Max}) range.", argumentName);
+        }
+        if (MaxInboundUnidirectionalStreams < 0 || MaxInboundUnidirectionalStreams > ushort.MaxValue)
+        {
+            throw new ArgumentOutOfRangeException($"{nameof(QuicConnectionOptions.MaxInboundUnidirectionalStreams)} should be within [0, {ushort.Max}) range.", argumentName);
+        }
+        if (IdleTimeout < TimeSpan.Zero && IdleTimeout != Timeout.InfiniteTimeSpan)
+        {
+            throw new ArgumentOutOfRangeException(nameof(QuicConnectionOptions.IdleTimeout), SR.net_quic_timeout_use_gt_zero);
+        }
+        if (DefaultStreamErrorCode < 0 || DefaultStreamErrorCode > QuicDefaults.MaxErrorCodeValue)
+        {
+            throw new ArgumentOutOfRangeException($"{nameof(QuicConnectionOptions.DefaultStreamErrorCode)} should be within [0, {QuicDefaults.MaxErrorCodeValue}) range.", argumentName);
+        }
+        if (DefaultCloseErrorCode < 0 || DefaultCloseErrorCode > QuicDefaults.MaxErrorCodeValue)
+        {
+            throw new ArgumentOutOfRangeException($"{nameof(QuicConnectionOptions.DefaultCloseErrorCode)} should be within [0, {QuicDefaults.MaxErrorCodeValue}) range.", argumentName);
+        }
+    }
 }
 
 /// <summary>
 /// Options for client (outbound) <see cref="QuicConnection" />.
 /// </summary>
 public sealed class QuicClientConnectionOptions : QuicConnectionOptions
 {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuicClientConnectionOptions"/> class.
+    /// </summary>
+    public QuicClientConnectionOptions()
+    {
+        MaxInboundBidirectionalStreams = QuicDefaults.DefaultClientMaxInboundBidirectionalStreams;
+        MaxInboundUnidirectionalStreams = QuicDefaults.DefaultServerMaxInboundUnidirectionalStreams;
```

Comment:
Possible copy-paste error?

```suggestion
        MaxInboundUnidirectionalStreams = QuicDefaults.DefaultClientMaxInboundUnidirectionalStreams;
```

Reply:
Fixed